tests/lib/Repository/Validator/TargetContentValidatorTest.php

```diff
@@ -0,0 +1,112 @@
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\Core\Repository\Validator;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\Core\FieldType\ValidationError;
+use eZ\Publish\Core\Repository\Values\ContentType\ContentType;
+use Ibexa\Core\Repository\Validator\TargetContentValidator;
+use PHPUnit\Framework\TestCase;
+
+final class TargetContentValidatorTest extends TestCase
+{
+    /** @var \eZ\Publish\API\Repository\ContentService|\PHPUnit_Framework_MockObject_MockObject */
+    private $contentService;
+
+    /** @var \eZ\Publish\API\Repository\ContentTypeService|\PHPUnit_Framework_MockObject_MockObject */
+    private $contentTypeService;
+
+    /** @var \Ibexa\Core\Repository\Validator\TargetContentValidator */
+    private $targetContentValidator;
+
+    public function setUp(): void
+    {
+        $this->contentService = $this->createMock(ContentService::class);
+        $this->contentTypeService = $this->createMock(ContentTypeService::class);
+
+        $this->targetContentValidator = new TargetContentValidator(
+            $this->contentService,
+            $this->contentTypeService
+        );
+    }
+
+    public function testValidateWithValidContent(): void
+    {
+        $contentId = 2;
+        $allowedContentTypes = ['article'];
+
+        $this->setupContentTypeValidation($contentId);
+
+        $validationError = $this->targetContentValidator->validate($contentId, $allowedContentTypes);
+
+        self::assertNull($validationError);
+    }
+
+    public function testValidateWithInvalidContentType(): void
+    {
+        $contentId = 2;
+        $allowedContentTypes = ['folder'];
+
+        $this->setupContentTypeValidation($contentId);
+
+        $validationError = $this->targetContentValidator->validate($contentId, $allowedContentTypes);
+
+        self::assertInstanceOf(ValidationError::class, $validationError);
+    }
+
+    private function setupContentTypeValidation(int $contentId): void
+    {
+        $contentTypeId = 55;
+        $contentInfo = new ContentInfo(['id' => $contentId, 'contentTypeId' => $contentTypeId]);
+        $contentType = new ContentType(['identifier' => 'article']);
+
+        $this->contentService
+            ->expects($this->once())
+            ->method('loadContentInfo')
+            ->with($contentId)
+            ->willReturn($contentInfo);
+
+        $this->contentTypeService
+            ->expects($this->once())
+            ->method('loadContentType')
+            ->with($contentInfo->contentTypeId)
+            ->willReturn($contentType);
+    }
+
+    /**
+     * @param mixed $id
+     *
+     * @dataProvider providerForInvalidContentIdentifiers
+     */
+    public function testValidateWithInvalidContentId($id): void
```

Comment:
Use strict types please, the PHPDoc is wrong, it never should be mixed.
```suggestion
     * @dataProvider providerForInvalidContentIdentifiers
     */
    public function testValidateWithInvalidContentId(int $id): void
```